spec/statements.md

```diff
@@ -1017,142 +1204,173 @@ It is a compile-time error for a return statement to appear in a finally block (
+- Otherwise, if the try block or a catch block of S encloses the throw point and if S has a finally block, control is transferred to the finally block. If the finally block throws another exception, processing of the current exception is terminated. Otherwise, when control reaches the end point of the finally block, processing of the current exception is continued.
 
 <!-- -->
 
--   If an exception handler was not located in the current function invocation, the function invocation is terminated, and one of the following occurs:
+- If an exception handler was not located in the current function invocation, the function invocation is terminated, and one of the following occurs:
 
 <!-- -->
 
--   If the current function is non-async, the steps above are repeated for the caller of the function with a throw point corresponding to the statement from which the function member was invoked.
+- If the current function is non-async, the steps above are repeated for the caller of the function with a throw point corresponding to the statement from which the function member was invoked.
 
--   If the current function is async and task-returning, the exception is recorded in the return task, which is put into a faulted or cancelled state as described in §15.15.2.
+- If the current function is async and task-returning, the exception is recorded in the return task, which is put into a faulted or cancelled state as described in §15.15.2.
 
--   If the current function is async and void-returning, the synchronization context of the current thread is notified as described in §15.15.3.
+- If the current function is async and void-returning, the synchronization context of the current thread is notified as described in §15.15.3.
 
 <!-- -->
 
--   If the exception processing terminates all function member invocations in the current thread, indicating that the thread has no handler for the exception, then the thread is itself terminated. The impact of such termination is implementation-defined.
+- If the exception processing terminates all function member invocations in the current thread, indicating that the thread has no handler for the exception, then the thread is itself terminated. The impact of such termination is implementation-defined.
 
 ## The try statement
 
 The try statement provides a mechanism for catching exceptions that occur during execution of a block. Furthermore, the try statement provides the ability to specify a block of code that is always executed when control leaves the try statement.
 
-[]{#Grammar_try_statement .anchor}try-statement:\
-*try* block catch-clauses\
-*try* block catch-clauses~opt~ finally-clause
+[#Grammar_try_statement .anchor](#Grammar_try_statement .anchor)try-statement:\
```

Comment:
This should be a single ANTLR block starting with this line and extending to line 1298